controllers/machinehealthcheck_controller.go

```diff
@@ -237,31 +236,50 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, cluster *c
 			EventRemediationRestricted,
 			"Remediation restricted due to exceeded number of unhealthy machines (total: %v, unhealthy: %v, maxUnhealthy: %v)",
 			totalTargets,
-			totalTargets-currentHealthy,
+			m.Status.CurrentHealthy,
 			m.Spec.MaxUnhealthy,
 		)
+		for _, t := range append(healthy, unhealthy...) {
+			if err := t.patchHelper.Patch(ctx, t.Machine); err != nil {
```

Comment:
I don't think I follow. I know what you mean about when remediation is restricted, but I don't follow about setting the remediated condition. We can't set it to false, since it hasn't been remediated, and setting it to true seems incorrect. I could see that we should add another condition, maybe on the MHC itself, to report that remediation is restricted.

Reply:
If I'm understanding the original comment correctly, @fabriziopandini is suggesting to be more explicit when marking the health check as false by populating Reason and Message. For example, if a node condition timeout has been reached, we have the ability to communicate that to the user.

Reply:
My intention was to have 
MachineOwnerRemediatedCondition=False with reason RemediationRestricted

Reply:
I think this is a separate feature, I don't want to expand the scope of this any further than it already has gotten

Reply:
Agreed, let's file an issue for what @fabriziopandini is asking for and discuss it there

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/3163